skupper-router.spec.rpkg

```diff
@@ -0,0 +1,156 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# This .spec file uses `rpkg` (https://pagure.io/rpkg-util) to provide
+# pleasant user experience to developers.
+#
+#  Command                  Description
+# `rpkg srpm`              Creates a *.src.rpm file in /tmp/rpkg (exact path is printed)
+# `rpkg local --nocheck`   Builds a *.rpm for your system in /tmp/rpkg (exact path is printed)
+#
+# See `man rpkg` for more commands.
+
+# https://bugzilla.redhat.com/show_bug.cgi?id=1668512#c19
+%undefine __cmake_in_source_build
+
+%global proton_minimum_version 0.34.0
+%global libwebsockets_minimum_version 3.0.1
+%global libnghttp2_minimum_version 1.33.0
+
+# https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/
+%define __python3 /usr/bin/python3.9
+
+Name:          skupper-router
+Version:       {{{ git_dir_version }}}
+Release:       2.0.0%{?dist}
+Summary:       Router for Skupper.io
+License:       ASL 2.0
+URL:           https://skupper.io
+VCS:           {{{ git_dir_vcs }}}
+
+BuildRequires: gcc
+BuildRequires: gcc-c++
+BuildRequires: cmake
+# https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_basic_system_settings/assembly_installing-and-using-python_configuring-basic-system-settings#proc_installing-python-3_assembly_installing-and-using-python
+# https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/packaging_and_distributing_software/index#con_common-macros-for-python-3-rpms_assembly_packaging-python-3-rpms
+%{?fedora:BuildRequires: python3.9}
+%{?rhel:BuildRequires: python39-devel}
```

Comment:
This is attempt to use Python 3.9 everywhere, as per https://github.com/skupperproject/skupper-router/issues/48.

```
ModuleNotFoundError: No module named 'proton'
```

That happens because `python3-qpid-proton` RPM installs Proton only for the system-default Python. I should've thought of that! Using static embedded build of Proton will not completely help, because the spec that @mcressman has now still needs the RPM package to do, ..., things.

Furthermore, I am not completely positive on the exact procedure to use the 3.9 Python in Fedora and CentOS. I asked on StackOverflow ;P https://stackoverflow.com/questions/71782123/how-do-i-install-python39-rpm-macros-for-fedora-35

Reply:
This can be resolved. We will build Qpid Proton statically as part of this RPM build anyways, which means that the python proton part can be built as well. I am not sure where to put it, though, so that it is out of the way of the user of the system, but still findable for the router. Seems to me that nonstandard location then requires some modifications in the router so that it will look there. (It can be small changes to PYTHONPATH setting, or note the `-I` switch for skrouterd... feels messy

Reply:
Attempt to use Python 3.9 everywhere is considered abandoned, because it was decided so.

Reply:
(I'll bring it up again when I'll know how to implement it cleanly; It should become easier when Proton build is embedded in building this RPM)